recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,187 @@
+                self.requires("libiconv/1.15")
+
+        if self.options.with_backend_postgresql:
+            self.requires("libpq/11.5")
+
+        if self.options.with_backend_db2:
+            # FIXME: add db2 support
+            raise ConanInvalidConfiguration("db2 is not (yet) available on CCI")
+            self.requires("db2/x.y.z")
+
+        if self.options.with_backend_firebird:
+            # FIXME: add firebird support
+            raise ConanInvalidConfiguration("firebird is not (yet) available on CCI")
+            self.requires("firebird/x.y.z")
+
+        if self.options.with_backend_oracle:
+            # FIXME: add oracle support
+            raise ConanInvalidConfiguration("oracle is not (yet) available on CCI")
+            self.requires("oracle/x.y.z")
+
+        if self.options.with_boost:
+            self.requires("boost/1.73.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("soci-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["BINDIR"] = os.path.join(self.package_folder, "bin")
+        self._cmake.definitions["LIBDIR"] = os.path.join(self.package_folder, "lib")
+        self._cmake.definitions["LIB_SUFFIX"] = ""
+
+        self._cmake.definitions["SOCI_SHARED"] = self.options.shared
+        self._cmake.definitions["SOCI_STATIC"] = not self.options.shared
+        self._cmake.definitions["SOCI_TESTS"] = False
+
+        self._cmake.definitions["WITH_BOOST"] = self.options.with_boost
+        self._cmake.definitions["SOCI_EMPTY"] = self.options.with_backend_empty
+        self._cmake.definitions["WITH_SQLITE3"] = self.options.with_backend_sqlite3
+        self._cmake.definitions["WITH_MYSQL"] = self.options.with_backend_mysql
+        self._cmake.definitions["WITH_ODBC"] = self.options.with_backend_odbc
+        self._cmake.definitions["WITH_POSTGRESQL"] = self.options.with_backend_postgresql
+
+        self._cmake.definitions["WITH_DB2"] = self.options.with_backend_db2
+        self._cmake.definitions["WITH_FIREBIRD"] = self.options.with_backend_firebird
+        self._cmake.definitions["WITH_ORACLE"] = self.options.with_backend_oracle
+
+        self._cmake.configure()
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        if self.options.with_backend_sqlite3:
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_FOUND", "SQLITE3_FOUND")
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_INCLUDE_DIRS", "SQLITE3_INCLUDE_DIRS")
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_LIBRARIES", "SQLITE3_LIBRARIES")
+        if self.options.with_backend_mysql:
+            os.rename("Findlibmysqlclient.cmake", "FindMySQL.cmake")
```

Comment:
We probably need here a PR to existing `libmysqlclient` to change the name for `cmake_find_package[_multi]` at least. It should be `MySQL` according to https://gitlab.kitware.com/cmake/community/-/wikis/contrib/modules/FindMySQL

Reply:
I would like to change mysql recipe, but would it not break backward compatibility?

Reply:
I would like to change mysql recipe, but would it not break backward compatibility?

Reply:
The previous name was the default and plain wrong.
Also, there is currently no way to use more then 2 names.
There is a certain backward compatibility with respect to versions available and names of options, but by my knowledge not for cmake_find_package(_multi) names (if they were wrong before).
So I would go ahead and fix the mysql receipe.

Reply:
Updated the recipe, but there is now a dependency on #1669 for Findmysqlclient.cmake => FindMySQL.cmake renaming.

Reply:
We try not to break and try to think everything in advance, but recipes need to evolve and improve, if we had the same stability commitment in ConanCenter as we have for the Conan client any change will be eventually be blocked and recipes will become really complex trying to keep every single working scenario. A nightmare! 😅 

As long as a PR improves recipe quality we need to merge it... and if it breaks some other recipes we'll try to fix them asap, especially if they are really common ones.

Thanks for the changes, let's wait on #1669

Reply:
#1669 has been merged